packages/flutter_tools/lib/src/doctor.dart

```diff
@@ -403,13 +404,16 @@ class IntelliJValidatorOnMac extends IntelliJValidator {
     }
 
     try {
-      for (FileSystemEntity dir in fs.directory('/Applications').listSync()) {
-        if (dir is Directory) {
-          checkForIntelliJ(dir);
-          if (!dir.path.endsWith('.app')) {
-            for (FileSystemEntity subdir in dir.listSync()) {
-              if (subdir is Directory)
-                checkForIntelliJ(subdir);
+      var checkDirectories = [ '/Applications' , path.join(homeDirPath, 'Applications') ];
```

Comment:
Another way to do it and clean up the existing code a bit / reduce the indentation might be:
```
Iterable<FileSystemEntity> appDirs = <String>['/Applications', path.join(...)]
    .map((String dir) => fs.directory(dir).listSync())
    .expand((FileSystemEntity e) => e)
    .where((FileSystemEntity e) => e is Directory);
for (FileSystemEntity dir in appDirs) {
  ...
}
```

Totally optional, up to you if you'd like to play with that a bit.